### PR TITLE
Working watchdog plugin executable by INI

### PR DESCRIPTION
**Pull request by ribab**
Watchdog plugin made to be executable from an INI file.
Semaphores added to prevent watchdog from throwing interrupt while program can't catch the interrupt.

### CODE REVIEW

**Comment by rhc54** [comment]
I think it would really help if you put some explanation at the top of the sequential plugin explaining what all these calls to "acquire" mean. As things stand, I have no idea what is going on (since I am not intimately familiar with the Semaphore library), and therefore have no way to review this PR.

**Comment by ribab** [comment]
@rhc54 The semaphores are there to prevent the watchdog timer from generating an exception while the main thread is not within the context that catches exceptions.

I'll add the comments to describe their function in the code

File: pylib/Tools/Executor/sequential.py [hunk 1]
```diff
@@ -66,16 +66,24 @@ def execute(self, testDef):
         status = 0
         only_reporter = False
         stageOrder = testDef.loader.stageOrder
+
+        # Holding a semaphore while in transition between plugins
+        # so async threads don't interrupt in the wrong context
+        testDef.plugin_trans_sem.acquire()
```
Comment (ribab) [inline pylib/Tools/Executor/sequential.py]: Hi @rhc54 -- I added this comment to describe the semaphores more. Does this help?

**Comment by ribab** [comment]
Hi @rhc54 -- I added this comment to describe the semaphores more. Does this help?

```
+        # Holding a semaphore while in transition between plugins
+        # so async threads don't interrupt in the wrong context
+        testDef.plugin_trans_sem.acquire()
```

**Pull request merged**